core/src/main/java/org/pentaho/platform/dataaccess/datasource/wizard/service/impl/DSWDatasourceServiceImpl.java

```diff
@@ -579,13 +581,8 @@ public DatasourceDTO deSerializeModelState( String dtoStr ) throws DatasourceSer
       component.validate( PentahoSessionHolder.getSession(), null );
       component.setAction( PMDUIComponent.ACTION_LIST_MODELS );
       Document document = component.getXmlContent();
-      List<DefaultElement> modelElements = document.selectNodes( "//model_name" ); //$NON-NLS-1$
 
-      ArrayList<String> datasourceNames = new ArrayList<String>();
-      for ( DefaultElement element : modelElements ) {
-        datasourceNames.add( element.getText() );
-      }
-      return datasourceNames;
+      return ( (List<Node>) document.selectNodes( "//model_name" ) ).stream().map( x -> x.getText() ).collect( toList() );
```

Comment:
While correct, this diminishes readability. Not sure this was warranted.

Reply:
See: https://github.com/pentaho/data-access/pull/1021